llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1336,6 +1332,30 @@ void VPlanTransforms::addExplicitVectorLength(VPlan &Plan) {
   NextEVLIV->insertBefore(CanonicalIVIncrement);
   EVLPhi->addOperand(NextEVLIV);
 
+  forAllHeaderPredicates(Plan, [VPEVL](VPInstruction &Mask) {
```

Comment:
An alternative approach would be to scan the recipes inside the loop, replacing every widen memory recipe with its vp counterpart, where header masks are replaced with null. Before doing so, collect all header masks into a set, possibly via forAllHeaderPredicates(), which could be simplified into collectAllHeaderPredicates(), until a designated recipe can be use directly.

Where is `collectUsersRecursively()` defined?